src/fitnesse/components/PluginsClassLoader.java

```diff
@@ -2,45 +2,54 @@
 
 import java.io.File;
 import java.lang.reflect.Method;
+import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.List;
 
 
+/**
+ * Update the current thread class path with jars foundin a "plugins" directory.
+ */
 public class PluginsClassLoader {
 
-  private File pluginsDirectory;
+  public static ClassLoader loadPlugins(String rootPath) throws Exception {
+    File pluginsDirectory = new File(rootPath, "plugins");
 
-  public PluginsClassLoader(String rootPath) {
-    pluginsDirectory = new File(rootPath, "plugins");
+    URL[] urls = urlsForPlugins(pluginsDirectory);
+    if (urls.length > 0) {
+      return new URLClassLoader(urls, ClassLoader.getSystemClassLoader());
+    } else {
+      return ClassLoader.getSystemClassLoader();
+    }
   }
 
-  public void addPluginsToClassLoader() throws Exception {
-    if (pluginsDirectory.exists())
+  private static URL[] urlsForPlugins(File pluginsDirectory) throws Exception {
+    List<URL> urls = new ArrayList<>();
+
+    if (pluginsDirectory.exists() && pluginsDirectory.isDirectory())
       for (File plugin : pluginsDirectory.listFiles())
         if (plugin.getName().endsWith("jar"))
-          addItemsToClasspath(plugin.getCanonicalPath());
+          urls.addAll(toUrls(plugin.getCanonicalPath()));
+
+    return urls.toArray(new URL[urls.size()]);
   }
 
-  public static void addItemsToClasspath(String classpathItems) throws Exception {
+  private static List<URL> toUrls(String classpathItems) throws Exception {
     final String separator = File.pathSeparator;
     String currentClassPath = System.getProperty("java.class.path");
     System.setProperty("java.class.path", currentClassPath + separator + classpathItems);
```

Comment:
Shouldn't this reading/parsing/updating of system property be in a different method? (Called after generating the full list of urls.)
Actually updating the property should probably only be done in once the class loader is created and made active. Should it be moved to `ClassUtils.setClassLoader()`, if it is called with a URLClassLoader it could call `getURLs()` and update the property with that result?

Reply:
Yes. I actually thought of removing this line altogether (no more system properties updating).

Reply:
Fixed (removed this method)